polyfill/lib/parse.mjs

```diff
@@ -0,0 +1,15 @@
+import { absolute as regex } from './regex.mjs';
+
+export function parse(isoString) {
+  const match = regex.exec(isoString);
+  if (!match) throw new RangeError(`Invalid ISO 8601 string: ${isoString}`);
+  return {
+    absolute: match[0],
+    dateTime: match[1],
+    date: match[2],
+    yearMonth: `${match[3]}-${match[4]}`,
+    monthDay: `${match[4]}-${match[5]}`,
+    time: match[6],
+    zone: match[13]
```

Comment:
I think it would be better to have the offset and the IANA name available as separate properties on this object, rather than one `zone: "+01:00[Europe/Berlin]"` property. That way, you can more easily make an intelligent decision on what to do if they don't agree. (With just one property, you'd have to parse the square brackets yourself.)